src/DynamoCore/Core/CommandLineRunner.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo;
+using ProtoCore;
+using Dynamo.Models;
+using System.Xml;
+using System.Threading;
+using System.IO;
+
+namespace Dynamo.Core
+{
+    public struct CommandLineArguments
+    {
+        internal static CommandLineArguments FromArguments(string[] args)
```

Comment:
Yes, definitely use the Command Line Parser Library for this.  It will make your life much nicer.  Remember to include the license along with it, though.